internal/client/garden/client.go

```diff
@@ -483,6 +486,30 @@ func (g *clientImpl) GetCloudProfile(ctx context.Context, name string) (*gardenc
 	return cloudProfile, nil
 }
 
+func (g *clientImpl) CheckUserRoles(ctx context.Context) (bool, error) {
+	review := &authorizationv1.SelfSubjectAccessReview{
+		Spec: authorizationv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationv1.ResourceAttributes{
+				Verb:     "get",
+				Resource: "secrets",
+			},
+		},
+	}
+
+	err := g.c.Create(ctx, review)
```

Comment:
in go-test,
```
••••E1019 07:48:16.551599   23955 ssh.go:56] "could not get node names from shoot" err="failed to create self subject access review  \"\" is invalid: metadata.name: Required value: name is required"

------------------------------
• [FAILED] [0.574 seconds]
SSH Command ValidArgsFunction [It] should find nodes based on their prefix
/tmp/build/a94a8fe5/pull-request-gardener.gardenctl-v2-master-pr/pkg/cmd/ssh/ssh_test.go:665

  [FAILED] Expected
      <[]string | len:0, cap:0>: nil
  to have length 1
  In [It] at: /tmp/build/a94a8fe5/pull-request-gardener.gardenctl-v2-master-pr/pkg/cmd/ssh/ssh_test.go:672 @ 10/19/23 07:48:16.552
------------------------------
•••••

Summarizing 1 Failure:
  [FAIL] SSH Command ValidArgsFunction [It] should find nodes based on their prefix
  /tmp/build/a94a8fe5/pull-request-gardener.gardenctl-v2-master-pr/pkg/cmd/ssh/ssh_test.go:672
```

@petersutter Do you happen to know how to pass the `go-test` in here

Reply:
Hi @tedteng, instead of asking for permission, ask for forgiveness ;) which means basically: try to get the seed client (machine objects case). If it fails with permission error, ignore it and continue with the regular flow. This will reduce some complexity of the code

Reply:
Thanks, 😃  That could be one solution. I will try changing it to reduce the complexity here